resolver.go

```diff
@@ -180,7 +180,7 @@ func (r *Resolver) resolve(ctx context.Context, qname, qtype string, depth int)
 func (r *Resolver) iterateParents(ctx context.Context, qname, qtype string, depth int) (RRs, error) {
 	chanRRs := make(chan RRs, MaxNameservers)
 	chanErrs := make(chan error, MaxNameservers)
-	ctx, cancel := context.WithCancel(ctx)
+	child_ctx, cancel := context.WithCancel(ctx)
```

Comment:
Variables shouldn't be named in the snake_case format. So let's rename this to something like `childCtx`.

Reply:
Oh silly mistake, i fixed it.